awsconfigs/apps/jupyter-web-app/configs/secrets-pod-default.yaml

```diff
@@ -0,0 +1,47 @@
+apiVersion: kubeflow.org/v1alpha1
+kind: PodDefault
+metadata:
+  name: add-aws-secret
+  labels:
+    add-aws-secret : "true"
+spec:
+  desc: add-aws-secret
+  selector:
+    matchLabels:
+      add-aws-secret: "true"
+  env:
+  - name: AWS_ACCESS_KEY_ID
+    valueFrom:
+      secretKeyRef:
+        name: mlpipeline-minio-artifact
+        key: accesskey
+  - name: AWS_SECRET_ACCESS_KEY
+    valueFrom:
+      secretKeyRef:
+        name: mlpipeline-minio-artifact
+        key: secretkey
+  - name: username
+    valueFrom:
+      secretKeyRef:
+        name: mysql-secret
+        key: username
+  - name: password
+    valueFrom:
+      secretKeyRef:
+        name: mysql-secret
+        key: password
+  - name: host
+    valueFrom:
+      secretKeyRef:
+        name: mysql-secret
+        key: host
+  - name: database
+    valueFrom:
+      secretKeyRef:
+        name: mysql-secret
+        key: database
+  - name: port
+    valueFrom:
+      secretKeyRef:
+        name: mysql-secret
+        key: port
```

Comment:
After doing an end-to-end test I discovered that this part does not work since `mysql-sercret` is in `kubeflow` namespace (not user namespace eg: `kubeflow-user-example-com`) then we cannot use it as this `PodDefault` should be applied to the user's namespace (eg: `kubeflow-user-example-com`).  
  
It works for `mlpipeline-minio-artifact` secret since this secret is created in the user's namespace.  
  
Any thoughts/ideas/feedback would be greatly appreciated.

Reply:
I want to take a step back and ask why does user need access to database? If this was a common usecase I would expect the profile controller to take care of this


regarding the solution, I think you are in the right direction. the correct path would be to use the secrets provider to sync secrets from Secrets manager similar to how it is done in kubeflow namespace.

the secrets-manager component also makes this easy, user just need to change the namespace in kustomization.yaml and kustomize build and apply it

you are right accessing cross namespace wont work.

Reply:
You bring up a good point, the motivation is important.  
**Use Cases :**  
Here are the use cases that this PR aims to solve :  
- Be able to use `boto3` in a notebook and specify credentials without hard coding them into the notebook (access the credentials via env variables)
- Be able to use `ml-metadata` in a notebook (eg: to explore metadata, compare metrics) and specify the [credentials needed](https://github.com/google/ml-metadata/blob/master/g3doc/get_started.md) by using env variables
- Be able to use `ml-metadata` to query metadata during a pipeline run by using something similar to [`use_aws_secret`](https://kubeflow-pipelines.readthedocs.io/en/latest/_modules/kfp/aws.html) from kfp sdk. This is an important use case for me as we want to use `ml-metadata` to get back information about the current model in production, compare metrics to generate visualization during a pipeline run.
- Be able to use [`use_aws_secret`](https://kubeflow-pipelines.readthedocs.io/en/latest/_modules/kfp/aws.html) from kfp sdk to authenticate pipeline to push artifacts to s3 (artifacts are pushed automatically by kfp sdk v2).  
  
**About The Soltuion**  
>he correct path would be to use the secrets provider to sync secrets from Secrets manager similar to how it is done in kubeflow namespace.  
> the secrets-manager component also makes this easy, user just need to change the namespace in kustomization.yaml and kustomize build and apply it   

I'm not super familiar with `SecretProviderClass`, from my understanding you are suggesting we could create a new "empty" secret named `mysql-secret` in the user namespace, then we'd update the namespace for `secrets-mount.yaml`, `secrets-manager.yaml` and  `kustomization.yaml` to the user's namespace, then we could do kustomize build and apply and the secrets provider will be able to update the `mysql-secret` with the correct values in the user namespace?

Reply:
I have applied your suggestion @surajkota , see the updated changes.